gitlab/resource_gitlab_user_impersonation_token_test.go

```diff
@@ -0,0 +1,239 @@
+						Active:    true,
+						Scopes:    []string{"api", "read_user"},
+						Revoked:   false,
+						ExpiresAt: &iso_d,
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGitlabUserImpersonationToken_import(t *testing.T) {
+	var token gitlab.ImpersonationToken
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGitlabUserImpersonationTokenDestroy,
+		Steps: []resource.TestStep{
+			// Create a user and impersonation token
+			// gitlab_user is already tested as part of `resource_gitlab_user`
+			{
+				Config: testAccGitlabUserImpersonationTokenConfig(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabUserImpersonationTokenExists("gitlab_user_impersonation_token.bar", &token),
+					testAccCheckGitlabUserImpersonationTokenAttributes("gitlab_user_impersonation_token.bar", &token, &testAccCheckGitlabUserImpersonationTokenExpectedAttributes{
+						Name:    fmt.Sprintf("Token bar %d", rInt),
+						Active:  true,
+						Scopes:  []string{"api"},
+						Revoked: false,
+					}),
+				),
+			},
+			{
+				ResourceName:      "gitlab_user_impersonation_token.bar",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// the API can't serve these fields, so ignore them
+				ImportStateVerifyIgnore: []string{"token"},
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabUserImpersonationTokenExists(n string, token *gitlab.ImpersonationToken) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		tokenID := rs.Primary.ID
+		if tokenID == "" {
+			return fmt.Errorf("No token ID is set")
+		}
+
+		userID := rs.Primary.Attributes["user_id"]
+		if userID == "" {
+			return fmt.Errorf("No user ID is set")
+		}
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		usertoken := strings.Split(rs.Primary.ID, ":")
+		userId, _ := strconv.Atoi(usertoken[0])
+		tokenId, _ := strconv.Atoi(usertoken[1])
+		gotToken, _, err := conn.Users.GetImpersonationToken(userId, tokenId, nil)
+		if err != nil {
+			return err
+		}
+		*token = *gotToken
+		return nil
+	}
+}
+
+func testAccGitlabUserImpersonationTokenDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*gitlab.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "gitlab_user_impersonation_token" {
+			continue
+		}
+
+		usertoken := strings.Split(rs.Primary.ID, ":")
+		userId, _ := strconv.Atoi(usertoken[0])
+		tokenId, _ := strconv.Atoi(usertoken[1])
+
+		token, resp, err := conn.Users.GetImpersonationToken(userId, tokenId, nil)
+		if err == nil {
+			if token != nil && fmt.Sprintf("%d", token.ID) == rs.Primary.ID && !token.Revoked {
+				return fmt.Errorf("Impersonation token still exists ")
+			}
+		}
+		if resp.StatusCode != 404 {
+			return err
+		}
```

Comment:
I'm having trouble understanding the `if resp.StatusCode != 404` bit here. What was your intent? Would this work instead?

```suggestion
		if err != nil {
			return err
		}
		if !token.Revoked {
			return errors.New("impersonation token still exists")
		}
```